File: restoGhid-backend/src/main/java/com/example/licentaRestaurante/entities/Review.java
package com.example.licentaRestaurante.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.*;

import javax.persistence.*;
import java.time.LocalDate;

@Getter
@Setter
@Entity
@Table(name = "review")
@AllArgsConstructor
@RequiredArgsConstructor
public class Review {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "date")
    private LocalDate date;

    @Column(name = "score")
    private Double score;

    @Column(name = "content")
    private String content;

    @Column(name = "reply")
    private String reply;

    @Column(name = "username")
    private String username;

    @Column(name = "id_user")
    private Long idUser;

    @Column(name = "id_restaurant")
    private String idRestaurant;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_user", insertable = false, updatable = false)
    @JsonIgnore
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_restaurant", insertable = false, updatable = false)
    @JsonIgnore
    private Restaurant restaurant;
}

File: restoghid-android/app/src/main/java/com/example/licenta/models/Review.kt
package com.example.licenta.models

import com.google.gson.annotations.SerializedName
import java.time.LocalDate
import java.time.ZonedDateTime

data class Review(
    @SerializedName("id")
    val id: Long? = null,

    @SerializedName("idUser")
    val userId: Long? = null,

    @SerializedName("idRestaurant")
    val restaurantId: String? = null,

    @SerializedName("content")
    val content: String,

    @SerializedName("score")
    val score: Double,

    @SerializedName("username")
    val username: String?= null,

    @SerializedName("reply")
    val reply: String? = null,

    @SerializedName("date")
    val date: String? = null
)
